### DISCUSSION TITLE
[Question]: Require login every time refresh

### DISCUSSION THREAD

**Discussion by SailFlorve** [danny-avila/LibreChat#572] [Troubleshooting] [answered]
### What is your question?

why every time I refresh the web, login required? what should I config?

### More Details

and if there is any way to disable user auth system?

### What is the main subject of your question?

User System/OAuth

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
> why every time I refresh the web, login required? what should I config?

This is not the behavior of the app. It's probably due to your cookie handling on your browser.

> and if there is any way to disable user auth system?

**Yes,** you can disable user registration once you are signed up. You can't disable the authentication system altogether: https://docs.librechat.ai/features/user_auth_system.html#disable-user-registration

**Reply by brandoncorbin** [reply]
FYI Every browser that I'm trying it on (Brave - with sheild turned off), Chrome, Safari all require a login after a page refresh.  The link provided above is no longer available.

**Reply by berry-13** [reply]
here's the correct link https://docs.librechat.ai/install/user_auth_system.html#disable-user-registration
are you on the latest version of LibreChat? because I have no problems of any kind even with brave (shield turn on). Try to provide more information (example: do you use a domain? do you use docker? etc...)

**Reply by anes300** [reply]
I experience the very same thing on my setup. I'm running the application through Docker (compose) on Ubuntu Server. I've looked through any configuration that could be made in the browser without luck. I tested the application on my Windows machine with Docker Desktop, and I don't experience the issue there.

@danny-avila, do you have any idea what to look for here?

**Reply by SailFlorve** [reply]
I already found the solution. When I use IP address or http protocol to access LibreChat, then I need to login again every time I open it. This is solved by configuring https. If I use IP address, it requests login everytime.

**Reply by anes300** [reply]
@SailFlorve, how did you configure https if I may ask?

**Reply by SailFlorve** [reply]
nginx

**Reply by danny-avila** [reply]
@anes300 i don’t use https and I don’t have to login every time.

have you set a SESSION_EXPIRY and REFRESH_TOKEN_EXPIRY?

https://docs.librechat.ai/install/user_auth_system.html#first-time-setup

and make sure you use docker-compose build after changing env file

make your session expiry the equivalent of one day in milliseconds

**Reply by SailFlorve** [reply]
this should not be the case. I deployed using Docker, but when accessing via IP, it does require a login every time. This was the case even before the SESSION_EXPIRYfields were implemented, and they work fine with https. so it’s not related to them. I’m not sure what the reason is.

**Reply by danny-avila** [reply]
yes I'm not sure, I can't reproduce and I almost exclusively use localhost (http)

**Reply by anes300** [reply]
My apologies, I wasn't specific enough. This only happens if I try to connect to the application from another client and not when doing it on localhost. It seems like it's not saving the JWT token on the client. I have indeed setup the SESSION_EXPIRY and REFRESH_TOKEN_EXPIRY to the default values according to the .env.example.

**Reply by SailFlorve** [reply]
Yeah. It happens when remote client access by IP, not localhost.

**Reply by fuegovic** [reply]
> Yeah. It happens when remote client access by IP, not localhost.

I don't know how to fix it but I was able to reproduce

I host LibreChat on a home server on my local network

- if I access LibreChat through the url `https://example.com` everything works as expected
- if I access LibreChat through the server local network address  `http://192.168.xx.xx:3080/` I need to re-login evvery time I refresh the page

**Comment by danny-avila** [comment, marked as answer]
I know what's going on now after thinking about this more. It's related to this: https://github.com/danny-avila/LibreChat/discussions/471

@fuegovic @anes300 @SailFlorve

The cookies are set to secure in "production" mode, which is the default when you run the server. According to [expressjs docs](https://expressjs.com/en/advanced/best-practice-security.html#use-cookies-securely):
> secure - Ensures the browser only sends the cookie over HTTPS.

I forget this because they are still sent in a localhost environment. Without cookies, you need to login every time.

After I merge this PR, you can run the cookies in the insecure mode by running `npm run backend:dev` to start the app. For docker, you change the NODE_ENV in the compose file to `development`

Related PR: https://github.com/danny-avila/LibreChat/pull/1088

Please confirm this works

**Reply by davecrab** [reply]
I'm having this same issue, I've set the NODE_ENV=development in both the .env and docker-compose.override files and rebuilt the image but still getting the same issue. Any ideas? 

I'm just accessing this locally on my network via http via safari. Happens both on iOS and Mac

**Reply by danny-avila** [reply]
> via http

you should access it via HTTPS. 

> NODE_ENV

If you are sure about running in the insecure mode which can expose you to many vulnerabilities, it should be `NODE_ENV=CI`

**Reply by davecrab** [reply]
Ok I'll look at setting up https with caddy, thanks for the response!

**Reply by itoudium** [reply]
Hello. I also encountered the same problem but was able to resolve it.
I am using docker-compose to expose it on my home local network.
It turns out that specifying NODE_ENV was not effective because it is directly defined in package.json, rather than being provided in .env or docker-compose.yml. Instead, I found a workaround by merging the following content into `docker-compose.override.yml`:

```yaml:docker-compose.override.yml
services:
  api:
    command: npm run backend:dev
```

Thank you for the wonderful product.

**Comment by sn4901** [comment]
> Hello. I also encountered the same problem but was able to resolve it. I am using docker-compose to expose it on my home local network. It turns out that specifying NODE_ENV was not effective because it is directly defined in package.json, rather than being provided in .env or docker-compose.yml. Instead, I found a workaround by merging the following content into `docker-compose.override.yml`:
> 
> ```miniyaml
> services:
>   api:
>     command: npm run backend:dev
> ```
> 
> Thank you for the wonderful product.

Having the OP issue on my Unraid server using docker. Does anyone know how to apply the above fix in Unraid?

**Reply by fuegovic** [reply]
It depends on how you deployed using unraid, if you used portainer in unraid you can easily modify the compose file directly in portainer to add this. If you used to the community app, I'm not certain.

**Reply by sn4901** [reply]
I did use a community app, yeah. No worries! Hopefully someone comes across this and knows.

**Reply by dadaclonefly** [reply]
@snab43
Did you ever figure this out? I'm having the same issue in Unraid using the community app version

**Comment by freddong** [comment]
`npm run backend:dev` stopped working.

I have to use
```
    command: node api/server/index.js
```
with 
```
    environment:
      - NODE_ENV=development
```

**Reply by danny-avila** [reply]
What web app keeps you logged in forever? On `localhost` you're not going to have refresh token handling, and the ways around it are insecure and not good practice

**Reply by svatyvabin** [reply]
It works, edit docker-compose.override.yml file , add code blow for api service:
```yaml
    command: node api/server/index.js
    environment:
      - NODE_ENV=development
```

the `npm run backend:dev` command or .env file `NODE_ENV=development` both not work.

**Comment by ghost** [comment]
You can simply setup Nginx as reverse proxy and create self signed certiciates. This will work no matter what and is independent if you run LC via docker or npm.

**Reply by danny-avila** [reply]
Agreed! Also worth mentioning [Ngrok](https://ngrok.com/) for the less technical. It's also great for development.

**Comment by rhonyabdullah** [comment]
I still having this issue, I'm running on docker for my local machine so I don't care anything, at this moment I just want it keeps logged in if I close my tab or refresh the browser. Here is my current config which not working:

`docker-compose.override.yml`
```
services:
  # USE LIBRECHAT CONFIG FILE
  api:
    command: npm run backend:dev
    environment:
      - NODE_ENV=development
    volumes:
      - type: bind
        source: ./librechat.yaml
        target: /app/librechat.yaml
```
`.env`
```
#==================================================#
#                      Others                      #
#==================================================#
#   You should leave the following commented out   #

NODE_ENV=development

# E2E_USER_EMAIL=
# E2E_USER_PASSWORD=
```

Any help would be appreciated, thanks in advance.

Why I need this `refresh` ?
: Because currently I'm heavily testing multiple models pulling and removing on the models. Testing those models requires a refresh each time I'm changing the model array (models : default: [ ... ] ) on `librechat.yaml`

**Reply by danny-avila** [reply]
This matches what has already been discussed, but whether it takes depend on what commands you are running, and if you are properly cycling down the containers after these changes.

That said, what commands are you running?

**Reply by rhonyabdullah** [reply]
I didn't notice the command, was just using the `restart` button from docker desktop

![Screen Shot 2025-06-12 at 19 32 35 PM](https://github.com/user-attachments/assets/e74aaf3c-be03-42f5-a12a-43d257ceccb1)

Surprisingly it works now when I'm using `docker compose up` from **terminal** and I see this log `[nodemon] starting node api/server/index.js`

Now refreshing the browser doesn't take me to login screen.

Thanks for your insight @danny-avila !
